src/Compiler/TypedTree/TypedTree.fs

```diff
@@ -4556,28 +4556,37 @@ type TupInfo =
 type Measure = 
 
     /// A variable unit-of-measure
-    | Var of typar: Typar
+    | Var of typar: Typar * range: range
 
     /// A constant, leaf unit-of-measure such as 'kg' or 'm'
-    | Const of tyconRef: TyconRef
+    | Const of tyconRef: TyconRef * range: range
 
     /// A product of two units of measure
-    | Prod of measure1: Measure * measure2: Measure
+    | Prod of measure1: Measure * measure2: Measure * range: range
 
     /// An inverse of a units of measure expression
     | Inv of measure: Measure
 
     /// The unit of measure '1', e.g. float = float<1>
-    | One
+    | One of range: range
 
     /// Raising a measure to a rational power 
-    | RationalPower of measure: Measure * power: Rational
+    | RationalPower of measure: Measure * power: Rational * range: range
```

Comment:
The newly added code in TypedTreeOps is assigning range0 to the range anyway, is there a codepath assigning a meaningful value to it (and different from measure.Range) ?

Reply:
Same here we do not need this extra range. We can use the measure range